src/test/integration/IntegrationBase.t.sol

```diff
@@ -351,6 +347,234 @@ abstract contract IntegrationBase is IntegrationDeployer {
+
+            // if (isAllocation) ...
+            if (prevAlloc.currentMagnitude < allocateParams.newMagnitudes[i]) {
+                uint256 delta = allocateParams.newMagnitudes[i] - prevAlloc.currentMagnitude;
+                
+                if (completed) {
+                    // FIXME: fix completed case
+                    // assertEq(
+                    //     curMagnitude.encumbered, 
+                    //     prevMagnitude.encumbered, 
+                    //     string.concat(err, " (encumbered magnitude)")
+                    // );
+
+                    // assertEq(
+                    //     curMagnitude.allocatable, 
+                    //     prevMagnitude.allocatable, 
+                    //     string.concat(err, " (allocatable magnitude)")
+                    // );
+                } else {
+                    assertEq(
+                        curMagnitude.encumbered, 
+                        prevMagnitude.encumbered + delta, 
+                        string.concat(err, " (encumbered magnitude)")
+                    );
+
+                    assertEq(
+                        curMagnitude.allocatable, 
+                        prevMagnitude.allocatable - delta, 
+                        string.concat(err, " (allocatable magnitude)")
+                    );
+                }
+            } else {
+                uint256 delta = prevAlloc.currentMagnitude - allocateParams.newMagnitudes[i];
+
+                // if deallocated, and completed
+                if (completed) {
+                    assertEq(
+                        curMagnitude.encumbered - delta, 
+                        prevMagnitude.encumbered, 
+                        string.concat(err, " (encumbered magnitude)")
+                    );
+
+                    assertEq(
+                        curMagnitude.allocatable, 
+                        prevMagnitude.allocatable + delta, 
+                        string.concat(err, " (allocatable magnitude)")
+                    );
+                } else {
+                    assertEq(
+                        curMagnitude.encumbered, 
+                        prevMagnitude.encumbered, 
+                        string.concat(err, " (encumbered magnitude)")
+                    );
+
+                    assertEq(
+                        curMagnitude.allocatable, 
+                        prevMagnitude.allocatable, 
+                        string.concat(err, " (allocatable magnitude)")
+                    );
+                }
+            }
+
+            // Max magnitude should not have mutated.
+            assertEq(
+                curMagnitude.max, 
+                prevMagnitude.max, 
+                string.concat(err, " (max magnitude)")
+            );
+        }
+    }
+
+    function assert_Snap_Allocations_Slashed(
+        SlashingParams memory slashingParams,
+        OperatorSet memory operatorSet,
+        bool completed,
```

Comment:
What does completed mean in this context? When would this value be false?

Reply:
It's supposed to differentiate whether the allocation was completed before slashed or not.